engine/src/main/java/com/sandboni/core/engine/finder/bcel/visitors/http/JavaxControllerClassVisitor.java

```diff
@@ -30,12 +31,10 @@ public void visitMethod(Method method) {
     @Override
     public Stream<Link> start(JavaClass jc, Context c) {
         controllerPath = null;
-        AnnotationEntry path = getAnnotation(jc.getConstantPool(), jc::getAnnotationEntries, Annotations.JAVAX.PATH.getDesc(), Annotations.JAVAX.OPTIONS.getDesc());
+        Optional<AnnotationEntry> path = getAnnotation(jc.getConstantPool(), jc::getAnnotationEntries, Annotations.JAVAX.PATH.getDesc(), Annotations.JAVAX.OPTIONS.getDesc());
 
-        if (path != null) {
-            controllerPath = path.getAnnotationType().contains(Annotations.JAVAX.PATH.getDesc()) ?
-                    getAnnotationParameter(path, "value") : "";
-        }
+        path.ifPresent(annotationEntry -> controllerPath = annotationEntry.getAnnotationType().contains(Annotations.JAVAX.PATH.getDesc()) ?
```

Comment:
instead of ifPresent -- better to map -- because you are returning value rather than just executing logic

Reply:
Noted for future improvement!